### PR TITLE
Submit registration properties to hubspot up front

### PR DESCRIPTION
**Pull request by esoergel**
https://dimagi-dev.atlassian.net/browse/HI-609
Reverting a few recent changes.  There was an issue with boolean property updates not appearing when submitted as part of this form update, so those properties were switched to be submitted separately: https://github.com/dimagi/commcare-hq/pull/24114
Then all properties were moved into that subsequent request, which was also moved to happen after the creation: https://github.com/dimagi/commcare-hq/pull/24146
These couple minor fixes were also reverted, since they no longer apply: https://github.com/dimagi/commcare-hq/pull/24203 https://github.com/dimagi/commcare-hq/pull/24207

I believe moving the property updates to a separate request failed due to a delay in the user being registered in hubspot - we submit the form, then immediately ping to confirm the user exists, so any delay would cause us to get a false miss.  Moving it back into the original form seems to do the trick.

The issue with the boolean properties was I believe one of encoding.  We're currently using [this API](https://developers.hubspot.com/docs/methods/forms/submit_form) which accepts a content type of `application/x-www-form-urlencoded`, so `True` doesn't do anything, but `"true"` does.  It looks like a better option for the future would be to switch to the [v3 api](https://developers.hubspot.com/docs/methods/forms/submit_form_v3) which supports JSON and does input validation and helpful error responses, but I'm punting on that now since it only supports updating properties defined in the form you're submitting, whereas the older API seems to support arbitrary property updates.

I also added tests, which I found useful in understanding how this code worked, but I could be persuaded to drop them - not sure how useful they'll be on an ongoing basis.

### CODE REVIEW

File: corehq/apps/analytics/tests/test_hubspot.py [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+    HUBSPOT_SIGNUP_FORM_ID,
+    track_web_user_registration_hubspot,
+)
+
+
+@override_settings(ANALYTICS_IDS={'HUBSPOT_API_ID': '1234'})
+@mock.patch('corehq.apps.analytics.tasks.requests.get', mock.MagicMock())
+@mock.patch('corehq.apps.analytics.tasks.requests.post', mock.MagicMock())
+@mock.patch('corehq.apps.analytics.tasks._send_hubspot_form_request')
+class TestSendToHubspot(TestCase):
```
Comment (nickpell) [inline corehq/apps/analytics/tests/test_hubspot.py]: Yay tests 💪

**Review by nickpell — APPROVED** [review]

File: corehq/apps/analytics/tasks.py [hunk 2]
```diff
@@ -234,19 +228,22 @@ def _send_form_to_hubspot(form_id, webuser, hubspot_cookie, meta, extra_fields=N
...
 
-        # these must be submitted after the form to ensure
-        # the contact has been created in hubspot
-        if extra_fields:
-            update_hubspot_properties_v2(webuser, extra_fields)
-
 
 @count_by_response_code(DATADOG_HUBSPOT_SENT_FORM_METRIC)
-def _send_hubspot_form_request(url, data):
+def _send_hubspot_form_request(hubspot_id, form_id, data):
```
Comment (calellowitz) [inline corehq/apps/analytics/tasks.py]: out of curiosity, why did you move this down here?

File: corehq/apps/analytics/tests/test_hubspot.py [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+    domain = 'test-send-to-hubspot'
+
+    def test_registration(self, _send_hubspot_form_request):
```
Comment (calellowitz) [inline corehq/apps/analytics/tests/test_hubspot.py]: Is this test supposed to test anything directly related to the bug, or just the general workflow?

**Review by calellowitz — APPROVED** [review]
Couple small questions but overall looks good

File: corehq/apps/analytics/tasks.py (hunk 2 above)
Comment (esoergel) [reply corehq/apps/analytics/tasks.py]: Seemed like a more appropriate place for it.  This way the function above doesn't need the URL, it just says "send this form with this data", without needing to worry about the details of assembling the URL and initializing the request.

 Plus, the URL used for this endpoint determines what parameters need to be passed to the request.  For instance, I attempted upgrading to v3 (more on that in the PR description), which would mean that `data` should be passed as JSON rather than HTTP form data.

File: corehq/apps/analytics/tests/test_hubspot.py (hunk 3 above)
Comment (esoergel) [reply corehq/apps/analytics/tests/test_hubspot.py]: Just the general workflow - I'm not 100% convinced it's doing anything useful at this point, but I found it really helpful when figuring out what was actually happening.  The first version of this did test the bug, but then when I rolled back the recent changes, that no longer applied.
